anta/tests/logging.py

```diff
@@ -0,0 +1,297 @@
+            return
+
+        self.logger.debug(f"self.instance_commands is: {self.instance_commands}")
+        command_output = cast(str, self.instance_commands[0].output)
+        self.logger.debug(f"dataset is: {command_output}")
+
+        pattern = rf"Logging source-interface '{intf}'.*VRF {vrf}"
+
+        if re.search(pattern, command_output):
+            self.result.is_success()
+        else:
+            self.result.is_failure(f"Source-interface '{intf}' is not configured in VRF {vrf}")
+
+
+class VerifyLoggingHosts(AntaTest):
+    """
+    Verifies logging hosts (syslog servers) for a specified VRF.
+
+    Expected Results:
+        * success: The test will pass if the provided syslog servers are configured in the specified VRF.
+        * failure: The test will fail if the provided syslog servers are NOT configured in the specified VRF.
+        * skipped: The test will be skipped if syslog servers or VRF are not provided.
+    """
+
+    name = "VerifyLoggingHosts"
+    description = "Verifies logging hosts (syslog servers) for a specified VRF."
+    categories = ["logging"]
+    commands = [AntaTestCommand(command="show logging | awk '/Trap logging/,/Sequence numbers/{if(/Sequence numbers/) exit; print}'", ofmt="text")]
+
+    @AntaTest.anta_test
+    def test(self, hosts: Optional[List[str]] = None, vrf: str = "default") -> None:
+        """
+        Run VerifyLoggingHosts validation.
+
+        Args:
+            hosts: List of hosts (syslog servers) IP addresses.
+            vrf: The name of the VRF to transport log messages. Defaults to 'default'.
+        """
+        if not hosts or not vrf:
+            self.result.is_skipped(f"{self.__class__.name} did not run because hosts or vrf were not supplied")
+            return
+
+        self.logger.debug(f"self.instance_commands is: {self.instance_commands}")
+        command_output = cast(str, self.instance_commands[0].output)
+        self.logger.debug(f"dataset is: {command_output}")
+
+        not_configured = []
+
+        for host in hosts:
+            pattern = rf"Logging to '{host}'.*VRF {vrf}"
+            if not re.search(pattern, command_output):
+                not_configured.append(host)
+
+        if not not_configured:
+            self.result.is_success()
+        else:
+            self.result.is_failure(f"Syslog servers {not_configured} are not configured in VRF {vrf}")
+
+
+class VerifyLoggingLogsGeneration(AntaTest):
+    """
+    Verifies if logs are generated.
+
+    Expected Results:
+        * success: The test will pass if logs are generated.
+        * failure: The test will fail if logs are NOT generated.
+    """
+
+    name = "VerifyLoggingLogsGeneration"
+    description = "Verifies if logs are generated."
+    categories = ["logging"]
+    commands = [
+        AntaTestCommand(command="send log level informational message ANTA VerifyLoggingLogsGeneration validation"),
+        AntaTestCommand(command="show logging | tail", ofmt="text"),
```

Comment:
could this trigger false positive if there are too many logs generated between two commands? (tail does not show ANTA log anymore ?)


Maybe something more like:

```
Leaf3#show logging last 30 s | grep ANTA
May 11 17:45:11 Leaf3 ConfigAgent: %SYS-6-LOGMSG_INFO: Message from arista on vty4 (10.95.70.233): ANTA VerifyLoggingLogsGeneration validation
```

Reply:
Agreed.